src/hotspot/os_cpu/bsd_x86/os_bsd_x86.cpp

```diff
@@ -510,6 +510,9 @@ JVM_handle_bsd_signal(int sig,
     // Handle ALL stack overflow variations here
     if (sig == SIGSEGV || sig == SIGBUS) {
       address addr = (address) info->si_addr;
+#ifdef __FreeBSD__
```

Comment:
I have no objection to this change as a temporary fix, but perhaps it should be a port patch until we find the root cause. I have noted that the same issue exists on FreeBSD/aarch64 as well. I'm hoping to find some time to dig into this in detail  this week.

Reply:
I think that's fair.  I can leave it as a patch in the FreeBSD ports tree for the moment while the investigation continues.